src/ui/atoms/babylonEngine/BabylonEngine.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useRef, useState } from 'react'
+import type { EngineOptions, Nullable } from '@babylonjs/core'
+import { Engine } from '@babylonjs/core'
+import { Canvas } from 'ui/atoms/canvas/Canvas'
+import type { Callback, DeepReadonly, StateHook } from 'superTypes'
+
+export type EngineCreatedCallback<STATE> = Callback<Engine, STATE>
+export type RenderCallback<STATE> = Callback<STATE, STATE>
+
+export type BabylonEngineProps<STATE> = DeepReadonly<{
```

Comment:
Now is better to type the `DeepReadonly` on the props instead of the type (so that it does not appear in the story)
```suggestion
export type BabylonEngineProps<STATE> = DeepReadonly<{
```

Reply:
🤔 I don't see the point of doing this for the storybook since it doesn't have any impact on it, as far as I can see. I'll change it but the code is currently quite inconsistent regarding the correct use of `DeepReadOnly`, and the `readonly` keyword on the type properties level.

Reply:
Fixed by [214684d](https://github.com/okp4/ui/pull/365/commits/214684d953005de2f63e66ac87dfc30e05609b1e).